### ISSUE TITLE
Communication's component validator does not propagate exceptions, obscuring the cause of the error

### ISSUE THREAD

**Issue by nikita2206** [magento/magento2#14555]
It was quite hard to understand the issue with my request schema until I went into [`Magento\Framework\Communication\Config\Validator`](https://github.com/magento/magento2/blob/2.2-develop/lib/internal/Magento/Framework/Communication/Config/Validator.php#L71) and changed exception instantiation from `new LogicException($message)` to `new LogicException($message, 0, $e)`. My issue was to do with this in the end: `Each getter must have a doc block`. It would be better if it was possible to see that error message out of the box, but instead I was only seeing `Request schema definition for topic "..." should reference existing service class. Given ...` (while the class obviously existed).

### Preconditions
Magento 2.2.5

### Steps to reproduce
1. A schema class:
```
class Dummy {
    private $foo;

    public function getFoo() { return $this->foo; }
    public function setFoo($foo) { $this->foo = $foo; }
}
```

2. Communication configuration (`communication.xml`):
```
<?xml version="1.0"?>
<config xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="urn:magento:framework:Communication/etc/communication.xsd">
    <topic name="foo_topic" request="Dummy" />
</config>
```

3. Run validation for communication configuration. It can be done either automatically if you're using Magento EE, or can be done manually by calling into `Magento\Framework\Communication\Config->getTopic("foo_topic")`

### Expected result
![e](https://i.imgur.com/5afNFLw.png)

### Actual result
![r](https://i.imgur.com/WdArj7C.png)

**Comment by ArtsiomBruneuski** [comment]
#sqashtoberfest

**Comment by ArtsiomBruneuski** [comment]
#squashtoberfest

**Comment by magento-engcom-team** [comment]
@ArtsiomBruneuski thank you for joining. Please accept team invitation [here](https://github.com/orgs/magento/invitation) and self-assign the issue.

**Comment by ArtsiomBruneuski** [comment]
@magento-engcom-team give me 2.3-develop instance

**Comment by magento-engcom-team** [comment]
Hi @ArtsiomBruneuski. Thank you for your request. I'm working on Magento 2.3-develop instance for you

**Comment by magento-engcom-team** [comment]
Hi @ArtsiomBruneuski, here is your [Magento instance](https://i-14555-2-3-develop.engcom.dev.magento.com).
Admin access: [https://i-14555-2-3-develop.engcom.dev.magento.com/admin](https://i-14555-2-3-develop.engcom.dev.magento.com/admin)
Login: `admin` Password: `123123q`
Instance will be terminated in up to 3 hours.

**Comment by sidolov** [comment]
Hi @nikita2206. Thank you for your report. 
The issue has been fixed in magento/magento2#18416 by @ArtsiomBruneuski in 2.3-develop branch
Related commit(s):
 - [9a1b8fd6401f872f5f5f12db15839135e4f38d27](https://github.com/magento/magento2/commit/9a1b8fd6401f872f5f5f12db15839135e4f38d27)
 - [4f017f89e2588aaa0df2714935d24abd9100e981](https://github.com/magento/magento2/commit/4f017f89e2588aaa0df2714935d24abd9100e981)
 - [6dd7115e953efb35c03c648889d12068b3fa7fce](https://github.com/magento/magento2/commit/6dd7115e953efb35c03c648889d12068b3fa7fce)
 - [335c2011bd536b908ab7b5e888f16614b0e2c12e](https://github.com/magento/magento2/commit/335c2011bd536b908ab7b5e888f16614b0e2c12e)
 - [7903cd5d130846efbead59f27955a3a5a439b620](https://github.com/magento/magento2/commit/7903cd5d130846efbead59f27955a3a5a439b620)
 - [dfda40787fda0b0abe58520a74b72820f2c7fed7](https://github.com/magento/magento2/commit/dfda40787fda0b0abe58520a74b72820f2c7fed7)
 - [b2621a5a3310ca0b8177f350370eb6b95d5d2f22](https://github.com/magento/magento2/commit/b2621a5a3310ca0b8177f350370eb6b95d5d2f22)
 - [7392ba6993b625968539f204109d2e3327a72eb3](https://github.com/magento/magento2/commit/7392ba6993b625968539f204109d2e3327a72eb3)
 - [b37d2d6fb3fed1d4c39ca24de413d25961fe8008](https://github.com/magento/magento2/commit/b37d2d6fb3fed1d4c39ca24de413d25961fe8008)
 - [9f1d4bfa763f23a5a25f14bae95455dca3e4628f](https://github.com/magento/magento2/commit/9f1d4bfa763f23a5a25f14bae95455dca3e4628f)

The fix will be available with the upcoming 2.3.1 release.

**Comment by sidolov** [comment]
Hi @nikita2206. Thank you for your report. 
The issue has been fixed in magento/magento2#18554 by @gelanivishal in 2.2-develop branch
Related commit(s):
 - [3109909d76b0402a941518bc9306697e4431ef54](https://github.com/magento/magento2/commit/3109909d76b0402a941518bc9306697e4431ef54)
 - [9afdfee251b7ef9098b3ae22e59cf0673c557bb3](https://github.com/magento/magento2/commit/9afdfee251b7ef9098b3ae22e59cf0673c557bb3)
 - [791609c7ec59700694609937deeecd1916fda2b6](https://github.com/magento/magento2/commit/791609c7ec59700694609937deeecd1916fda2b6)
 - [beeecafdb28a20b39e06c17554e49bbffa1f7632](https://github.com/magento/magento2/commit/beeecafdb28a20b39e06c17554e49bbffa1f7632)
 - [7e38326028a4d2419fa553723c4eef4bd4c90495](https://github.com/magento/magento2/commit/7e38326028a4d2419fa553723c4eef4bd4c90495)
 - [a20968db7a29a3b6f26a46979179851121eea336](https://github.com/magento/magento2/commit/a20968db7a29a3b6f26a46979179851121eea336)
 - [2f2cafee687d81d95854b1629ef8eec7c5a4e548](https://github.com/magento/magento2/commit/2f2cafee687d81d95854b1629ef8eec7c5a4e548)
 - [72b31c952619104e1b176fe9c03074437c8a58b9](https://github.com/magento/magento2/commit/72b31c952619104e1b176fe9c03074437c8a58b9)
 - [a7cdae0e4f80aa8596d4b3a3b958ae2b9938a346](https://github.com/magento/magento2/commit/a7cdae0e4f80aa8596d4b3a3b958ae2b9938a346)
 - [74c6bfd295569253d2a371bf2f0d11fda931c68c](https://github.com/magento/magento2/commit/74c6bfd295569253d2a371bf2f0d11fda931c68c)

The fix will be available with the upcoming 2.2.8 release.
